### ISSUE TITLE
xmlrpc.html doesn't describe how to return errors to the client

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10757]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| spiv reported|
|-|-|
|Trac ID|trac#1152|
|Type|enhancement|
|Created|2005-08-29 12:37:03Z|
|Branch|https://github.com/twisted/twisted/tree/xmlrpc-fault-doc-1152|

<details><summary>Searchable metadata</summary>

```
trac-id__1152 1152
type__enhancement enhancement
reporter__spiv spiv
priority__normal normal
milestone__ 
branch__branches_xmlrpc_fault_doc_1152 branches/xmlrpc-fault-doc-1152
branch_author__ 
status__closed closed
resolution__fixed fixed
component__web web
keywords__documentation documentation
time__1125319023000000 1125319023000000
changetime__1214406861000000 1214406861000000
version__ 
owner__ 
cc__spiv cc__itamarst cc__oubiwann
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

```
#!html
<pre>
twisted.web.xmlrpc.XMLRPC allows errors to be returned to the client by raising
xmlrpclib.Fault from a handler method, e.g.

    def xmlrpc_bang(self):
        """Sends an error to the client with faultCode 1234"""
        raise xmlrpclib.Fault(1234, 'Bang!')

twisted.web.soap.SOAPPublisher seems to handle SOAPpy.faultType in a similar
fashion.

The "Creating XML-RPC Servers and Clients with Twisted" howto (xmlrpc.html)
should document this aspect of writing XML-RPC and SOAP servers.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst set owner to @oubiwann|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="oubiwann's avatar" src="https://avatars.githubusercontent.com/u/894251?s=50" width="50" height="50">](https://github.com/oubiwann)|@oubiwann set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun commented|
|-|-|

(In [23989]) Branching to 'xmlrpc-fault-doc-1152'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

(In [23990]) talk about fault; fixup `__all__` a bit

refs [#10757](https://github.com/twisted/twisted/issues/10757)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

okay, the xmlrpc howto talks about fault now.  I didn't do anything with soap at all, since I don't think the xmlrpc howto is the right place to talk about soap, and hopefully someone will do something with zsi soon.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

grlgk soap soap soap

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

Okay, I almost went and updated the soap docs too, but you actually _cannot_ get a soap fault with `twisted.web.soap`, it's not supported.  I also can't tell if the server behaves correctly.  I don't know what the HTTP response code is supposed to be when indicating a soap fault.  I suspect it should not be 500.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_10"></a>|@therve set owner to @exarkun|
|-|-|

In the doc, you mean "p.fault()" instead of "s.fault()". The rest looks good, +1.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun commented|
|-|-|

(In [24020]) the variable is s

refs [#10757](https://github.com/twisted/twisted/issues/10757)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_12"></a>|@exarkun commented|
|-|-|

(In [24021]) replace > with &gt; where it is necessary

refs [#10757](https://github.com/twisted/twisted/issues/10757)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun set status to `closed`|
|-|-|

(In [24022]) Merge xmlrpc-fault-doc-1152

Author: exarkun
Reviewer: therve
Fixes: [#10757](https://github.com/twisted/twisted/issues/10757)

Add an example of generating a fault to the XMLRPC documentation.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_14"></a>|@exarkun commented|
|-|-|

(In [24024]) Merge xmlrpc-fault-doc-1152

Author: exarkun
Reviewer: therve
Fixes: [#10757](https://github.com/twisted/twisted/issues/10757)

Add an example of generating a fault to the XMLRPC documentation.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
